react/AddToCartButton.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { FC, useCallback } from 'react'
+import { useMutation } from 'react-apollo'
+import {
+  FormattedMessage,
+  injectIntl,
+  InjectedIntlProps,
+  defineMessages,
+} from 'react-intl'
+import { Button, Tooltip } from 'vtex.styleguide'
+import { OrderForm } from 'vtex.order-manager'
+import { useCssHandles } from 'vtex.css-handles'
+import { useRuntime } from 'vtex.render-runtime'
+import { addToCart as ADD_TO_CART } from 'vtex.checkout-resources/Mutations'
+import { useProductDispatch } from 'vtex.product-context/ProductDispatchContext'
+
+import { MapCatalogItemToCartReturn, compareObjects } from './utils'
+
+interface Props {
+  isOneClickBuy: boolean
+  available: boolean
+  disabled: boolean
+  customToastUrl: string
+  skuItems: MapCatalogItemToCartReturn[]
+  showToast: Function
+  allSkuVariationsSelected: boolean
+}
+
+interface OrderFormContext {
+  loading: boolean
+  orderForm: OrderForm | undefined
+  setOrderForm: (orderForm: Partial<OrderForm>) => void
+}
+
+const CSS_HANDLES = ['buyButtonText', 'buttonDataContainer']
+const CHECKOUT_URL = '/checkout/#/cart'
+
+const messages = defineMessages({
+  success: { id: 'store/addtocart.success', defaultMessage: '' },
+  duplicate: { id: 'store/addtocart.duplicate', defaultMessage: '' },
+  error: { id: 'store/addtocart.failure', defaultMessage: '' },
+  seeCart: { id: 'store/addtocart.see-cart', defaultMessage: '' },
+})
+
+const adjustItemsForMutationInput = (
+  newItems: MapCatalogItemToCartReturn[]
+): OrderFormItemInput[] => {
+  return newItems.map(item => ({
+    id: Number.parseInt(item.skuId),
+    index: item.index,
+    seller: item.seller,
+    quantity: item.quantity,
+    options: item.options,
+  }))
+}
+
+const AddToCartButton: FC<Props & InjectedIntlProps> = ({
+  intl,
+  isOneClickBuy,
+  available,
+  disabled,
+  skuItems,
+  customToastUrl,
+  showToast,
+  allSkuVariationsSelected = true,
+}) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  const {
+    orderForm,
+    setOrderForm,
+    loading,
+  }: OrderFormContext = OrderForm.useOrderForm()
+  const translateMessage = useCallback(
+    (id: string) => intl.formatMessage({ id }),
```

Comment:
You could receive a `MessageDescriptor` here instead of just an `id`. I don't remember the exact syntax, but I think is something like this

```suggestion
    (message: FormtattedMessage.MessageDescriptor) => intl.formatMessage(message),
```

Reply:
Niice!